data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoClientException.java

```diff
@@ -0,0 +1,17 @@
+package com.microsoft.azure.kusto.data.exceptions;
+
+public abstract class KustoClientException extends Exception {
+    public KustoClientException(String message) {
+        super(message);
+    }
+
+    public KustoClientException(String message, Exception exception) {
+        super(message, exception);
+    }
+
+    public abstract TriState isPermanent();
+
+    public static TriState triStateFromBool(boolean bool){
```

Comment:
Move this method to inside `TriState`, and then you can remove the prefix and name it `fromBool`.